internal/build_defs.bzl

```diff
@@ -55,11 +51,15 @@ def _compile_action(ctx, inputs, outputs, config_file_path):
     arguments = ["-p", config_file_path]
     mnemonic = "tsc"
 
+  outputs = non_externs_files
+  if not outputs:
```

Comment:
it's confusing me that you have other changes mixed in here. We already have some other PR open for this. Can you keep the changes non-overlapping?